File: C2-CMR/01-08-23/sliding_window.py


def main():
    arr = [1,-2,3,4,5,-34,-4,-32]
    k1 = 3
    i = 0 
    j = k1-1
    max_sum = float('-inf')
    while j < len(arr):
        sum = 0
        if i == 0:
            for k in range(i,j+1):
                sum+=arr[k]
            if sum > max_sum:
                max_sum = sum
            i+=1
            j+=1
            print(sum)
        else:
            sum = max_sum - arr[i-1] + arr[j]
            if sum > max_sum:
                max_sum = sum
            i+=1
            j+=1
    print(max_sum)


main()

File: CMR-EC/P1/19-09-2024/SlidingWindow.java
public class SlidingWindow {
    public static void main(String[] args) {
        int[] arr = {3,-2,4,5,-5,6,9,-1};
        int ws = 3;
        getMaximumSuarraySum(arr,ws);
    }

    private static void getMaximumSuarraySum(int[] arr, int ws) {
        int ans = Integer.MIN_VALUE, si = -1, ei = -1;
        int i = 0, j = ws-1;
        int previousSum = -1;
        while (j < arr.length) {
            int sum = 0;
            if (i == 0) {
                for(int k = i; k <=j;k++) sum+=arr[k];
                previousSum = sum;
            }else{
                sum = previousSum - arr[i-1] + arr[j];
                previousSum = sum;
            }
            i+=1;
            j+=1;
            if (sum > ans) {
                ans = sum;
                si = i;
                ei = j;
            }
        }
        System.out.println(si+" "+ei+" "+ans);
    }
}
